geoportailv3/static/js/mymaps/mymapsdirective.js

```diff
@@ -34,10 +35,136 @@ app.module.directive('appMymaps', app.mymapsDirective);
 
 
 /**
- * @ngInject
+ * @param {gettext} gettext Gettext service.
  * @constructor
  * @export
+ * @ngInject
+ */
+app.MymapsDirectiveController = function(gettext) {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.defaultTitle_ = gettext('Map without title');
+
+  /**
+   * Tells whether the 'modifying' modal window is open or not.
+   * @type {boolean}
+   * @export
+   */
+  this.modifying = false;
+
+  /**
+   * Tells whether the 'choosing a map' modal window is open or not.
+   * @type {boolean}
+   * @export
+   */
+  this.choosing = false;
+
+  /**
+   * The currently displayed map id.
+   * @type {?string}
+   * @export
+   */
+  this.mapId = null;
+
+  /**
+   * The currently displayed map title.
+   * @type {?string}
+   * @export
+   */
+  this.mapTitle = null;
+
+  /**
+   * The currently displayed map description.
+   * @type {?string}
+   * @export
+   */
+  this.mapDescription = null;
+
+  /**
+   * String to be used in the title field in the modifying window.
+   * Helps canceling modifications.
+   * @type {?string}
+   * @export
+   */
+  this.newTitle = null;
+
+  /**
+   * String to be used in the description field in the modifying window.
+   * Helps canceling modifications.
+   * @type {?string}
+   * @export
+   */
+  this.newDescription = null;
```

Comment:
I try to use the empty string as the initial value for strings, and `string` instead of `?string` as the type. But maybe that's not an option in your case…